### DISCUSSION TITLE
Using third party packages via <py-config>

### DISCUSSION THREAD

**Discussion by kenSpicer** [pyscript/pyscript#1839] [Q&A]
Jeff Glass pointed me to the <py-config> tag as the way to use packages that were not included with  pyscript so I wrote the following:

```html
<!DOCTYPE html>
<html lang="en">
<head>
  <meta charset="utf-8">
  <meta name="viewport" content="width=device-width, initial-scale=1">
  <title>Hello, World!</title>
  <link rel="stylesheet" href="../pyscript.css" />
  <script defer src="../pyscript.js"></script>
</head>
<body>
    <py-config>

      packages = ["PySimpleGUIQt"]

    </py-config>
    <py-script>
        print('Preparing to import PySimpleGUIQt')
        import PySimpleGUIQt as ps
        sg.Popup("This is a message from PySimpleGUIQt")
    </py-script>
</body>
</html>
```

pyscript yields a js exception reporting that it cannot saying that it cannot find the file.  Where must the file be for pyscript to find it?  The file is available in source folder, and the site-packages dir for Python.  What am i missing?

**Comment by WebReflection** [comment]
I'm afraid your original intent as posted code didn't go through so it's not clear what you are after but also we're realising PyScript next where things are either the same or different, surely better in general (what we hope) so you should really try to write a broken snippet correctly, here in GitHub, but also base your project on latest RC3 so that we can help forward, thanks!

**Reply by kenSpicer** [reply]
What i am after is using 'PySimpleGUIQt' in a pyscript app.  The package is in PyPi and installs easily with pip.  When I name it inside <py-config> or <py-env>  tags in my html code I get a js.exception saying that the file doesn't exist.  What don't I understand or am doing wrong?

**Reply by WebReflection** [reply]
like I've said, I wouldn't know as your original post has broken output ... could you please fix that? All I see is a broken post within its intent to provide valuable use case to think about, thanks!
![Screenshot from 2023-11-03 20-35-43](https://github.com/pyscript/pyscript/assets/85749/e3cb579c-052a-434c-9235-5641dc1cf1e4)

**Reply by WebReflection** [reply]
OK, I might have fixed that for you ... let me try again to understand the issue, thanks!

**Reply by WebReflection** [reply]
the gist of the issue seems to be:

```
ValueError: Can't find a pure Python 3 wheel for 'pyside2'.
```

so this is an upstream issue and there's nothing we can do ... this issue should probably land on *pyodide* project https://github.com/pyodide/pyodide

**Reply by JeffersGlass** [reply]
To expand on what Andrea said/linked, there are three kinds of packages you can install in Pyodide/PyScript:
  - Packages that have been [pre-built by the Pyodide team](https://pyodide.org/en/stable/usage/packages-in-pyodide.html), with their C/Rust/Fortran extensions pre-compiled to work in Web Assembly.
  - Pure Python packages that are available on [PyPI](https://pypi.org/); that is, packages which publish a wheel ending in -none-any.whl. (Some packages written only in Python do not publish Pure Python wheels; check the 'Download Files' section on PyPI to confirm.)
  - Pure Python wheels that you build yourself.

PySimpleGUIQt does seem to have a Pure Python wheel, but I suspect not all of its dependencies are.
